vividus-http-client/src/main/java/org/vividus/http/client/HttpClientConfig.java

```diff
@@ -33,36 +36,34 @@
 import org.apache.http.message.BasicHeader;
 import org.vividus.http.handler.HttpResponseHandler;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
 @SuppressWarnings("MethodCount")
 public class HttpClientConfig
 {
     private String baseUrl;
-    private String username;
-    private String password;
+    private AuthConfig authConfig;
     private AuthScope authScope;
     private Map<String, String> headers;
-    private boolean sslCertificateCheckEnabled = true;
-    private boolean sslHostnameVerificationEnabled = true;
+    private SslConfig sslConfig;
     private HttpClientConnectionManager connectionManager;
-    private int maxTotalConnections;
-    private int maxConnectionsPerRoute;
+    private Integer maxTotalConnections;
+    private Integer maxConnectionsPerRoute;
     private HttpRequestInterceptor lastRequestInterceptor;
     private HttpResponseInterceptor lastResponseInterceptor;
     private RedirectStrategy redirectStrategy;
-    private int connectionRequestTimeout = -1;
-    private int connectTimeout = -1;
-    private int socketTimeout;
+    private Integer connectionRequestTimeout;
+    private Integer connectTimeout;
+    private Integer socketTimeout;
     private CookieStore cookieStore;
-    private boolean skipResponseEntity;
+    private Boolean skipResponseEntity;
     private DnsResolver dnsResolver;
-    private boolean circularRedirectsAllowed;
+    private Boolean circularRedirectsAllowed;
     private String cookieSpec;
     private HttpRequestRetryHandler httpRequestRetryHandler;
     private ServiceUnavailableRetryStrategy serviceUnavailableRetryStrategy;
-    private boolean preemptiveAuthEnabled;
     private List<HttpResponseHandler> httpResponseHandlers;
 
-    public boolean hasBaseUrl()
+    public Boolean hasBaseUrl()
```

Comment:
what's the reason of this change?

Reply:
forgot to revert these changes after the right approach of merging things died :)